### PR TITLE
Homework 8

### PR DESCRIPTION
**Pull request by AbdurakhimovAziz**

### CODE REVIEW

File: client/src/app/components/accounts/accounts.component.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { Component, OnInit } from '@angular/core';
 import { Account } from 'src/app/shared/models/account';
 import { AccountsService } from 'src/app/shared/services/accounts.service';
+import { UserService } from 'src/app/shared/services/user.service';
```
Comment (AnastasiaHarost) [inline client/src/app/components/accounts/accounts.component.ts:5]: @UntilDestroy

File: client/src/app/components/accounts/accounts.component.ts [hunk 2]
```diff
@@ -9,11 +10,16 @@ import { AccountsService } from 'src/app/shared/services/accounts.service';
...
+    private accountsService: AccountsService,
+    private user: UserService
+  ) {}
 
   public ngOnInit(): void {
-    this.accountsService.getAll().subscribe((accounts: Account[]) => {
-      this.accounts = accounts;
-    });
+    this.accountsService
+      .getAll(this.user.getId())
```
Comment (AnastasiaHarost) [inline client/src/app/components/accounts/accounts.component.ts:20]: add please pipe(UntilDestoyed(this)) for all places where you subscribe to observables and ad @UntilDestriy to this components

File: client/src/app/components/header/header.component.ts [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
 })
-export class HeaderComponent {
-  constructor(private auth: AuthService, private router: Router) {}
+export class HeaderComponent implements OnInit {
+  user$!: Observable<User | null>;
```
Comment (AnastasiaHarost) [inline client/src/app/components/header/header.component.ts:14]: modifier

File: client/src/app/components/header/header.component.ts [hunk 4]
```diff
@@ -1,17 +1,30 @@
...
-  public logout(): void {
-    this.auth.logout();
-    this.router.navigateByUrl('/login');
+  constructor(
+    private userService: UserService,
+    private authService: AuthService,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
```
Comment (AnastasiaHarost) [inline client/src/app/components/header/header.component.ts:22]: modifier /return type for all func

File: client/src/app/components/transactions/transactions-item/transactions-item.component.html [hunk 5]
```diff
@@ -0,0 +1 @@
+<p>transactions-item works!</p>
```
Comment (AnastasiaHarost) [inline client/src/app/components/transactions/transactions-item/transactions-item.component.html:1]: remove

File: client/src/app/components/transactions/transactions.component.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  private currentAccount!: Account | null;
+
+  public transactions: Transaction[] = [];
+  constructor(
+    private transactionsService: TransactionsService,
+    private accountsService: AccountsService
+  ) {}
+
+  ngOnInit(): void {
+    this.accountsService.currentAccount$.subscribe((account) => {
```
Comment (AnastasiaHarost) [inline client/src/app/components/transactions/transactions.component.ts:22]: untilDestoyed

File: client/src/app/shared/services/accounts.service.ts [hunk 7]
```diff
@@ -1,25 +1,38 @@
...
 import { BASE_URL } from '../constants';
 import { Account } from '../models/account';
 
 @Injectable({
   providedIn: 'root',
 })
 export class AccountsService {
-  private accounts: Account[] = [];
-  constructor(private http: HttpClient) {}
+  private currentAccountSubject: BehaviorSubject<Account | null>;
```
Comment (AnastasiaHarost) [inline client/src/app/shared/services/accounts.service.ts:11]: private currentAccountSubject: BehaviorSubject<Account | null> = new BehaviorSubject<Account | null>(null);

File: client/src/app/shared/services/user.service.ts [hunk 8]
```diff
@@ -1,19 +1,32 @@
...
 
-  constructor() {}
+  constructor() {
+    this.userSubject = new BehaviorSubject<User | null>(this.getUser());
+    this.user$ = this.userSubject.asObservable();
+  }
 
   public setUser(user: User | null): void {
-    this.user = user;
+    console.log('setUser', user);
```
Comment (AnastasiaHarost) [inline client/src/app/shared/services/user.service.ts:18]: remove

File: client/src/app/components/transactions/transactions.component.ts (hunk 6 above)
Comment (AbdurakhimovAziz) [reply client/src/app/components/transactions/transactions.component.ts:23]: isn't it better to use async pipe or takeUntil than the external library?

Comment (AnastasiaHarost) [reply client/src/app/components/transactions/transactions.component.ts:23]: if you use async pipe, you won't have subscribe .ts file. You can use takeUntil

**Review by AnastasiaHarost — APPROVED** [review]

**Pull request merged**